Face/FindSimilar/Program.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.CognitiveServices.Vision.Face;
+using Microsoft.Azure.CognitiveServices.Vision.Face.Models;
+
+namespace FindSimilar
+{
+    public class Program
+    {
+        // Create a client.
+        public static string ApiKey = "ENTER YOUR KEY HERE";
+
+        public static IFaceClient Client = new FaceClient(new ApiKeyServiceClientCredentials(ApiKey))
+        {
+            Endpoint = "ENTER YOUR ENDPOINT HERE"
+        };
+
+        public static void Main(string[] args)
+        {
+            FindSimilar_in_FaceIds().Wait();
+            FindSimilar_in_FaceList().Wait();
+            FindSimilar_in_LargeFaceList().Wait();
+            Console.ReadLine();
+        }
+
+        public static async Task FindSimilar_in_FaceIds()
+        {
+            List<string> targetImageFileNames =
+                new List<string> { "Family1-Dad1.jpg", "Family1-Daughter1.jpg", "Family1-Mom1.jpg", "Family1-Son1.jpg", "Family2-Lady1.jpg", "Family2-Man1.jpg", "Family3-Lady1.jpg", "Family3-Man1.jpg" };
+            IList<Guid?> faceIds = new List<Guid?>();
+
+            foreach (var targetImageFileName in targetImageFileNames)
+            {
+                // Read image file.
+                using (FileStream stream = new FileStream(Path.Combine("Images", targetImageFileName), FileMode.Open))
+                {
+                    // Detect faces from image stream.
+                    IList<DetectedFace> detectedFaces = Client.Face.DetectWithStreamAsync(stream).Result;
+                    if (detectedFaces == null || detectedFaces.Count == 0)
+                    {
+                        Console.WriteLine($"[Error] No face detected from image `{targetImageFileName}`.");
+                        return;
+                    }
+
+                    Console.WriteLine($"{detectedFaces.Count} faces detected from image `{targetImageFileName}`.");
+                    if (detectedFaces[0].FaceId == null)
+                    {
+                        Console.WriteLine("[Error] Parameter `returnFaceId` of `DetectWithStreamAsync` must be set to `true` (by default) for verification purpose.");
+                        return;
+                    }
+
+                    // Add detected faceId to faceIds.
+                    faceIds.Add(detectedFaces[0].FaceId);
+                }
+            }
+
+            // Find similar example for find similar faces from faceIds with similar confidence.
+            IList<SimilarFace> similarResults = Client.Face.FindSimilarAsync(faceIds[0].Value, null, null, faceIds).Result;
```

Comment:
We should use same source/target images for all three samples.

Reply:
done